report/3.Theory/SyntaxTheory.tex

```diff
@@ -0,0 +1,178 @@
+	\end{itemize}
+\end{center}
+
+First the division is chosen for s.
+$s = xyz$, where $s = (((...b)))...$.
+Meaning that x consists only of left parentheses, y consists of a concatenation of left parentheses and elements from P and finally z consists only of right parentheses.
+
+The length of the string is $p < |s| < (p3)/2$, since $s = a^nbc^n$, where $n = p/2$ and $0 < |b| \leq p/2$.
+
+This means that the third condition is upheld since $|xy|$ cannot be greater than p.
+The second condition is also upheld since y consist of at least some left parentheses. 
+If the string does contain right parentheses, which it does. 
+Y cannot be of the length 0.
+
+The first condition is not upheld since if $i = 0$ the new string cannot contain an equal amount of left and right parentheses because of the same reason as to why the second condition is upheld.
+
+This concludes the proof by contradiction.
+Since HCl does not uphold the required property for it to be a regular language, it it not and the only automaton capable of recognizing the language is the NDPA described in the previous sections.
+
+\subsection{Types and Identifiers}
+So far the syntax of HCL has been properly defined as a CFL. 
+Meaning that all Variables, terminals and the transition rules for recognizing HCl has been described in a computational model as seen on page 
+%Indsæt sidetal for HCL CFG definition.
+and in appendix.
+%Indæst reference til appendix hvor NPDA for HCL er beskrevet.
+It is now relevant to define how the types and identifiers in HCL can be described or recognized by a computational model.
+The goal is to formally describe the languages $L_{none}$, $L_{bool}$, $L_{number}$, $L_{text}$ which are the type languages and the identifier language $L_{identifiers}$.
+
+An alphabet is a finite set of symbols, a string is a finite sequence of symbols and a language is a set of strings.
+To describe the languages mentioned above, the Alphabets used in the informal and formal notations needs to be defined. 
+These can be seen in Table 3.1.
+
+\begin{table}[!htb]
+	\centering
+	\begin{tabular}{|l|l|}
+		\hline
+		\textbf{Alphabet} & \textbf{Set}                            \\ \hline
+		T                 & \{true\}                                \\ \hline
+		F                 & \{false\}                               \\ \hline
+		$Z^+$             & The set of all positive integers        \\ \hline
+		$Z^-$             & The set of all negative integers        \\ \hline
+		R                 & The set of all real numbers             \\ \hline
+		D                 & \{$Z^+ \cup Z^-$\}                      \\ \hline
+		$S_p$             & The set of all special characters      \\ \hline
+		$L_l$             & The set of all lower-case letters       \\ \hline
+		$L_u$             & The set of all upper-case letters       \\ \hline
+		C                 & $S_p \cup L_l \cup L_u \cup D$          \\ \hline
+		$R_{eserved}$     & The set of all reserved keywords in HCL \\ \hline
```

Comment:
Change this to "all reserved keywords, specialchars and types"

Reply:
fixed